cli/src/pcluster/templates/login_nodes_stack.py

```diff
@@ -0,0 +1,205 @@
+        login_security_group,
+        stack_name,
+    ):
+        super().__init__(scope, id)
+        self._pool = pool
+        self._config = config
+        self._login_nodes_stack_id = id
+        self._shared_storage_infos = shared_storage_infos
+        self._shared_storage_mount_dirs = shared_storage_mount_dirs
+        self._shared_storage_attributes = shared_storage_attributes
+        self._login_security_group = login_security_group
+        self.stack_name = stack_name
+
+        self._add_resources()
+
+    def _add_resources(self):
+        self._vpc = ec2.Vpc.from_vpc_attributes(
+            self,
+            f"VPC{self._pool.name}",
+            vpc_id=self._config.vpc_id,
+            availability_zones=self._pool.networking.az_list,
+        )
+        self._login_nodes_pool_target_group = self._add_login_nodes_pool_target_group()
+        self._login_nodes_pool_load_balancer = self._add_login_nodes_pool_load_balancer(
+            self._login_nodes_pool_target_group
+        )
+
+        self._launch_template = self._add_login_nodes_pool_launch_template()
+        self._add_login_nodes_pool_auto_scaling_group()
+
+    def _add_login_nodes_pool_launch_template(self):
+        login_nodes_pool_lt_security_groups = get_login_nodes_security_groups_full(
+            self._login_security_group,
+            self._pool,
+        )
+        login_nodes_pool_lt_nw_interface = [
+            ec2.CfnLaunchTemplate.NetworkInterfaceProperty(
+                device_index=0,
+                interface_type=None,
+                groups=login_nodes_pool_lt_security_groups,
+                subnet_id=self._pool.networking.subnet_ids[0],
+            )
+        ]
+        return ec2.CfnLaunchTemplate(
+            self,
+            f"LoginNodeLaunchTemplate{self._pool.name}",
+            launch_template_name=f"{self.stack_name}-{self._pool.name}",
+            launch_template_data=ec2.CfnLaunchTemplate.LaunchTemplateDataProperty(
+                image_id=self._config.login_nodes_ami[self._pool.name],
+                instance_type=self._pool.instance_type,
+                key_name=self._pool.ssh.key_name,
+                metadata_options=ec2.CfnLaunchTemplate.MetadataOptionsProperty(
+                    http_tokens=get_http_tokens_setting(self._config.imds.imds_support)
+                ),
+                network_interfaces=login_nodes_pool_lt_nw_interface,
+                # user_data=
+                # block_device_mappings=
+                tag_specifications=[
+                    ec2.CfnLaunchTemplate.TagSpecificationProperty(
+                        resource_type="instance",
+                        tags=get_default_instance_tags(
+                            self.stack_name, self._config, self._pool, "LoginNode", self._shared_storage_infos
+                        )
+                        + [CfnTag(key=PCLUSTER_LOGIN_NODES_POOL_NAME_TAG, value=self._pool.name)]
+                        # + custom tags for instance
```

Comment:
Let's remove it instead of a comment.

Reply:
Done. BTW I also add new unit test for Security Groups.